src/Callback.php

```diff
@@ -9,6 +9,18 @@
 use function call_user_func_array;
 use function is_callable;
 
+/**
+ * @psalm-type OptionsProperty = array{
+ *     callback: callable|null,
+ *     callbackOptions: array<array-key, mixed>,
+ *     throwExceptions: bool,
+ * }
+ * @psalm-type OptionsArgument = array{
+ *     callback: callable,
+ *     callbackOptions?: array<array-key, mixed>,
+ *     throwExceptions?: bool,
+ * }&array<string, mixed>
```

Comment:
property does not need separate type declared imo.

Wouldn't it be preferred to copy options from the parent?

Psalm has unsealed array syntax:
```suggestion
 * @psalm-type OptionsArgument = array{
 *     callback: callable,
 *     callbackOptions?: array<array-key, mixed>,
 *     throwExceptions?: bool,
 *     ...<string, mixed>
 * }
```

Reply:
The reason I used another type was to make all options optional except for the callable. i.e. `throwExceptions?: bool` instead of `throwExceptions: bool`